### PR TITLE
[Character] Convert Load/Save of Character Buffs to Repositories

### PR DESCRIPTION
**Pull request by Kinglykrab**
# Notes
- Convert `LoadBuffs` to repositories.
- `SaveBuffs` was already using repositories, cleanup logic.

# Images
## Load
image
image
## Save
image
image

### CODE REVIEW

File: zone/entity.cpp [hunk 1]
```diff
@@ -1814,14 +1814,14 @@ void EntityList::DuelMessage(Mob *winner, Mob *loser, bool flee)
 	}
 }
 
-Client *EntityList::GetClientByName(const char *checkname)
```
Comment (Akkadius) [inline zone/entity.cpp:1817]: Really an out of scope change here - but we'll let it slide for this one. Cleaner for sure

File: zone/zonedb.cpp [hunk 2]
```diff
@@ -3110,160 +3110,150 @@ void ZoneDatabase::UpdateAltCurrencyValue(uint32 char_id, uint32 currency_id, ui
...
+		e.instrument_mod = buffs[slot_id].instrument_mod;
 
-		// add the buff to the vector
-		character_buffs.emplace_back(b);
+		v.emplace_back(e);
 	}
 
-	// insert the buffs into the database
-	if (!character_buffs.empty()) {
-		CharacterBuffsRepository::InsertMany(database, character_buffs);
```
Comment (Akkadius) [inline zone/zonedb.cpp:3170]: ReplaceMany ?

File: zone/zonedb.cpp [hunk 3]
```diff
@@ -3110,160 +3110,150 @@ void ZoneDatabase::UpdateAltCurrencyValue(uint32 char_id, uint32 currency_id, ui
...
-		int32 caston_z = Strings::ToUnsignedInt(row[13]);
-		int32 ExtraDIChance = Strings::ToUnsignedInt(row[14]);
-		uint32 instrument_mod = Strings::ToUnsignedInt(row[15]);
-
-		buffs[slot_id].spellid = spell_id;
-		buffs[slot_id].casterlevel = caster_level;
-
-		if (caster) {
-			buffs[slot_id].casterid = caster->GetID();
-			strcpy(buffs[slot_id].caster_name, caster->GetName());
```
Comment (Akkadius) [inline zone/zonedb.cpp:3222]: We changed strcpy methods again in a few places

File: zone/entity.cpp (hunk 1 above)
Comment (Kinglykrab) [reply zone/entity.cpp:1817]: Yeah, sorry. Only changed it because I was using it for this.

**Review by Akkadius — APPROVED** [review]

**Pull request merged**